<html><h3>:x: ci:test - 16 out of 18 jobs passed in 1 hour 31 minutes 30 seconds 568 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/ee1ef5f493f5e3b131d446d0ac609caf3c87b9e6">ee1ef5f493f5e3b131d446d0ac609caf3c87b9e6</a></p>16 out of 18 jobs PASSED<div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-19.liferay.com/job/test-subrepository-acceptance-pullrequest(7.1.x)/102/"><strike><strong>test-subrepository-acceptance-pullrequest(7.1.x)</strong></strike></a></li><li><a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/320/"><strike><strong>test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql57-jdk8</strong></strike></a></li></ul></div><details><summary><strong>16 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-3.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/389/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-3.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/390/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-assemble-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/540/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/409/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-20.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/328/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/543/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/541/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-15.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/149/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-19.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/386/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/410/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-18.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/258/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-15.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/150/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/542/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-19.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/387/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-18.liferay.com/job/test-subrepository-acceptance-pullrequest-dist(7.1.x)/22/">test-subrepository-acceptance-pullrequest-dist(7.1.x)/jdk8</a></li><li><a href="https://test-1-8.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/41/">test-subrepository-acceptance-pullrequest-validation(7.1.x)/jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-19.liferay.com/userContent/jobs/test-subrepository-acceptance-pullrequest(7.1.x)/builds/102/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/320/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql57-jdk8</a></h5><div><h6>Job Results:</h6><p>1805 Tests Passed.<br/>2 Tests Failed.</p></div><ol><li><div><a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/320//consoleText">AXIS_VARIABLE=0,label_exp=!master #320</a><ol><li><div><a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/320//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testCompletePayment">CommercePaymentEngineTest.testCompletePayment</a><pre><code>java.lang.AssertionError: expected:&lt;2&gt; but was:&lt;1&gt;
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testCompletePayment(CommercePaymentEngineTest.java:140)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.arquillian.junit.Arquillian$8$1.invoke(Arquillian.java:370)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquilli...</code></pre></div></li><li><div><a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/320//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testProcessPayment">CommercePaymentEngineTest.testProcessPayment</a><pre><code>java.lang.AssertionError: expected:&lt;2&gt; but was:&lt;1&gt;
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testProcessPayment(CommercePaymentEngineTest.java:202)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.arquillian.junit.Arquillian$8$1.invoke(Arquillian.java:370)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillia...</code></pre></div></li></ol></div></li></ol></div></li></ol></details></html>

--- a/commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/servlet/MercanetServlet.java
+++ b/commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/servlet/MercanetServlet.java
@@ -34,6 +34,7 @@ import com.liferay.portal.kernel.security.permission.PermissionCheckerFactoryUti
 import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
 import com.liferay.portal.kernel.servlet.PortalSessionThreadLocal;
 import com.liferay.portal.kernel.settings.GroupServiceSettingsLocator;
+import com.liferay.portal.kernel.util.GetterUtil;
 import com.liferay.portal.kernel.util.ParamUtil;
 import com.liferay.portal.kernel.util.Portal;
 import com.liferay.portal.kernel.util.StringUtil;
@@ -138,7 +139,7 @@ public class MercanetServlet extends HttpServlet {
 				if (!Objects.equals(parameterMap.get("responseCode"), "00")) {
 					String orderId = parameterMap.get("orderId");
 
-					Long commerceOrderId = Long.parseLong(orderId);
+					long commerceOrderId = GetterUtil.getLong(orderId);
 
 					String transactionReference = parameterMap.get(
 						"transactionReference");
@@ -186,16 +187,16 @@ public class MercanetServlet extends HttpServlet {
 				PaypageResponse paypageResponse = paypageClient.decodeResponse(
 					verifyMap);
 
-				ResponseData responseData = paypageResponse.getData();
-
-				ResponseCode responseCode = responseData.getResponseCode();
-
 				StringBundler transactionReferenceSB = new StringBundler(3);
 
 				transactionReferenceSB.append(commerceOrder.getCompanyId());
 				transactionReferenceSB.append(commerceOrder.getGroupId());
 				transactionReferenceSB.append(
 					commerceOrder.getCommerceOrderId());
+
+				ResponseData responseData = paypageResponse.getData();
+
+				ResponseCode responseCode = responseData.getResponseCode();
 
 				if (Objects.equals(responseCode.getCode(), "00") &&
 					Objects.equals(

